### PR TITLE
ENH: customizable Box Color in 3D view

### PR DESCRIPTION
**Pull request by kerim371**
Box in 3D view is an actor controlled by vtkMRMLViewDisplayableManager.
This manager also controls the renderer background color as well.
Thus it was pretty easy to introduce this feature in the same way.
`vtkMRMLViewNode` now have attribute `double BoxColor[3]`.
User is able to modify box color through vtkMRMLViewNode::SetBoxColor().

Probably it would be good to add the example to [Slicer script repository](https://slicer.readthedocs.io/en/latest/developer_guide/script_repository.html#change-3d-view-background-color):
```python
viewNode = slicer.app.layoutManager().threeDWidget(0).mrmlViewNode()
viewNode.SetBoxColor(1,0,0)
```

### CODE REVIEW

**Comment by kerim371** [comment]
Does ReadTheDocs and Slicer API docs checks didn't succed because of my changes?
Can't find information on that in the logs.

File: Libs/MRML/DisplayableManager/vtkMRMLViewDisplayableManager.cxx [hunk 1]
```diff
@@ -394,6 +410,16 @@ void vtkMRMLViewDisplayableManager::vtkInternal::UpdateAxisVisibility()
   this->External->RequestRender();
 }
 
+//---------------------------------------------------------------------------
+void vtkMRMLViewDisplayableManager::vtkInternal::UpdateAxisColor()
+{
+  double boxColor[3];
```
Comment (cpinter) [inline Libs/MRML/DisplayableManager/vtkMRMLViewDisplayableManager.cxx:416]: It's good practice to initialize any array with values. Above as well please.

Comment (kerim371) [reply Libs/MRML/DisplayableManager/vtkMRMLViewDisplayableManager.cxx:416]: Fixed

File: Libs/MRML/Core/vtkMRMLViewNode.cxx [hunk 2]
```diff
@@ -242,6 +249,13 @@ double* vtkMRMLViewNode::defaultBackgroundColor2()
   return backgroundColor2;
 }
 
+//------------------------------------------------------------------------------
+double* vtkMRMLViewNode::defaultBoxColor()
+{
+  static double boxColor[3] = {1.0, 0.0, 1.0};
```
Comment (lassoan) [inline Libs/MRML/Core/vtkMRMLViewNode.cxx:255]: Having a static member variable in a method is very obscure and could have unintended side effects. Also, naming of the method does not follow VTK conventions. Please change it to:

```
void vtkMRMLViewNode::GetDefaultBoxColor(double color[3])
{
  color[0] = 1.0;
  color[1] = 0.0;
  color[2] = 1.0;
}
```

I know that `defaultBackgroundColor` and `defaultBackgroundColor2` does not follow the naming conventions either but we'll fix that later.

Comment (kerim371) [reply Libs/MRML/Core/vtkMRMLViewNode.cxx:255]: Done

**Comment by jamesobutler** [comment]
@lassoan Do you have any additional comments here?

**Review by jamesobutler — APPROVED** [review]
Startup:
image
With example customization applied:
image

**Pull request merged**